### ISSUE TITLE
[serve] JS transform corrupts sourcemaps in bundled UMD scripts

### ISSUE THREAD

**Issue by web-padawan** [Polymer/tools#389]
## Description

As a library developer, I want to build the UMD bundle using `rollup` and include it in tests, which I run using `polymer serve`. As the bundle is already processed, and contains sourcemaps, I expect `polyserve` to recognize such scripts and not to apply any JS transform to them.

The problem here is that JS transform mangles some lines and messes up `debugger` statements.
Especially, some comments are not in the right place, some extra newlines added, etc.

### Expected outcome (polymer-cli 1.6.0)

Content of the file is untouched, `debugger` statement in the right place:

<img width="893" alt="screen shot 2018-05-15 at 21 02 23" src="https://user-images.githubusercontent.com/10589913/40075130-2e1ea66e-5884-11e8-8cd5-89d93f206154.png">

### Actual outcome (polymer-cli 1.7.2)

Content of the file is messed up. This is not easy to detect in the restored code in Chrome DevTools from the first look, but when running tests, code stops in a couple of lines below the breakpoint:

<img width="937" alt="screen shot 2018-05-15 at 19 28 12" src="https://user-images.githubusercontent.com/10589913/40075260-92819224-5884-11e8-96eb-7408f9793178.png">

I was able to reproduce the bug with shifting line numbers in Chrome 66 and Firefox 60.

### Steps to reproduce

1. Clone the repo
```sh
git clone https://github.com/web-padawan/cli-amd-sourcemap-issue
```
2. Install dependencies
```sh
npm install
```
3. Run the build
```sh
npm run build
```
4. Start dev server
```sh
npm start
```
5. Open Chrome DevTools

6. Navigate to http://127.0.0.1:8000/components/cli-amd-sourcemap-issue/test/

7. See the "sources" tab and check that code stopped below the breakpoint

8. Checkout to another branch

```sh
git checkout 1.6.0
```
9. Run npm install to switch `polymer-cli` version
```sh
npm install
```
10. Repeat steps 4. - 6. and see the breakpoint in the right place

### Processed code

In the gists below, you can find the content of the served files:
- [polymer-cli 1.6.0](https://gist.github.com/web-padawan/44224b9ffad7f2fd3abf6ede6c454413)
- [polymer-cli 1.7.2](https://gist.github.com/web-padawan/a71e2d2b63297f67f7728fec8157ee6d)

**Comment by rictic** [comment]
Oh! Good find, I was having a similar issue this weekend.

We still need to do some modifications to many libraries that will have source maps, so I think the best solution would be to either:

  - remove source map comments when we transform a js file
  - or update source maps when serving source maps for a transformed file

**Comment by web-padawan** [comment]
@rictic thanks for the feedback. How about disabling the JS transform for particular file, is that doable? Please note that if you check the gists from the "Processed code" section above, the only difference are changed newlines and shifted comments, so JS transform could skip that file at all.

I'm not using neither `import.meta` nor dynamic imports, so essentially no stuff to process by JS transform. But with the `transformModulesToAmd ` set to `'auto'` it still transforms the file.

**Comment by davidmaxwaterman** [comment]
Yeah, this frustrated me enough to switch to `python -m SimpleHTTPServer 8081&`, which works just fine.

**Comment by davidmaxwaterman** [comment]
FWIW, I'm not doing anything 'special' - ie not using rollup or webpack - and still find this problem. My 'development process' is a one-time `npm install && bower install` followed by the repeated edit/reload cycle.

The problem has made 'polymer serve' useless to me, and I have switched completely to `python -m SimpleHTTPServer 8081`.

I really think this should be getting more attention than it seems to be getting.

**Comment by web-padawan** [comment]
This regression happened before `async` / `await` support landed in `polyserve` which makes it even worse, as most of the code related to lit-element is now using those.

At the moment, I'm using webpack and karma for running tests, and do not have this issue.
My working setup can be found here: https://github.com/web-padawan/lit-components

**Comment by abdonrd** [comment]
Same issue in the PWA Starter Kit: https://github.com/Polymer/pwa-starter-kit/issues/250

**Comment by ankon** [comment]
Somewhat related: With https://github.com/Polymer/tools/pull/3334 it would be possible to make `polymer serve` produce sourcemaps.

**Comment by web-padawan** [comment]
This was quite a nasty issue back in the day.

Nowadays we can use `es-dev-server` which doesn't have this problem.

**Comment by davidmaxwaterman** [comment]
es-dev-server *does* have the problem where it totally gets the line numbers wrong, though. I've presumed that it is just because html files are now converted into js files, and you don't get to see the html in the dev tools any more...it's annoying :/
